instrumentation-genai/opentelemetry-instrumentation-google-genai/tests/generate_content/test_async_nonstreaming.py

```diff
@@ -0,0 +1,81 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import asyncio
+import logging
+import unittest
+
+from common.base import TestCase
+
+
+def create_valid_response(
+    response_text="The model response", input_tokens=10, output_tokens=20
+):
+    return {
+        "modelVersion": "gemini-2.0-flash-test123",
+        "usageMetadata": {
+            "promptTokenCount": input_tokens,
+            "candidatesTokenCount": output_tokens,
+            "totalTokenCount": input_tokens + output_tokens,
+        },
+        "candidates": [
+            {
+                "content": {
+                    "role": "model",
+                    "parts": [
+                        {
+                            "text": response_text,
+                        }
+                    ],
+                }
+            }
+        ],
+    }
+
+
+class TestGenerateContentAsyncNonstreaming(TestCase):
```

Comment:
could be worthwhile to add to the PR description why this is using a different recording approach than everything else. I'm sure you have reasons in you may have made in the comments, rather to bring it forward as this is special here. Notably, the rationale to intentionally make a different test process and why this is specific to google genai.

Reply:
I'm not sure what you have in mind in terms of the PR description, but I did just now update the documentation comments in the test code to address that question.

Reply:
I'm going to mark as resolved, since this is already called out in the code itself. And as it is an internal implementation detail that does not affect the public API exposed by this package, I don't know that it rises to the level of an additional call-out in the PR description.